File: oldSnacks/student_grade.py
number_of_students = int(input("How many students do you have? "))

number_of_subjects = int(input("How many subjects do they offer? "))

print("Saving >>>>>>>>>>>>>>>>>>>>>>>>>")

print("Saved sucessfully")

scores = [[0 for _ in range(number_of_subjects)] for _ in range(number_of_students)]

for counter in range(number_of_students):
	total_score = 0
	for count in range(number_of_subjects):
		while True:
			print(f"Entering score for student {counter+1}:")
			print(f"Enter score for subject {count+1}:")

			score = int(input())
			if 0 <= score <= 100:
				scores[counter][count] = score
				total_score += score
				break
			else:
				print("Invalid score. Please enter a value between 0 and 100.")

print("=============================================================")
print("STUDENT\t", end="")
for counter in range(number_of_subjects):
	print(f"\tSUB{counter+1:4}", end="")
print("\tTOT\tAVE\tPOS")
print("=============================================================")


for loop in range(number_of_students):
	total_score = 0
	average = 0.0

	print(f"STUDENT{loop+1}", end="")
	for looping in scores[loop]:
		print(f"\t{looping}", end="")
		total_score += looping

	average = total_score / number_of_subjects
	print(f"  {total_score:.2f}  {average:.2f}", end="")

	current_position = 1
	for continuation in range(number_of_students):
		if sum(scores[continuation]) > total_score:
			current_position += 1
	print(f"\t{current_position}")

print("================================================================")
subject_positions = [1] * number_of_subjects

for looped in range(number_of_subjects):
	passes = 0
	fails = 0
	for loops in range(number_of_students):
		if scores[loops][looped] >= 50:
			passes += 1
		else:
			fails += 1

	print(f"\nSubject {looped+1} Summary:")
	print(f"Number of Passes: {passes}")
	print(f"Number of Fails: {fails}")

print("\n================================================================\n")

File: oldSnacks/StudentGrade.java
import java.util.Arrays;
import java.util.Scanner;

public class StudentGrade {

    public static void main(String[] args) {

    Scanner scanner = new Scanner(System.in);

    System.out.println("How many students do you have? ");
    int noOfStudents = scanner.nextInt();

    System.out.println("How many subjects do they offer? ");
    int noOfSubjects = scanner.nextInt();

    System.out.println("Saving >>>>>>>>>>>>>>>>>>>>>>>>>");

    System.out.print("Saved sucessfully");

    System.out.println();

    
    int[][] scores = new int[noOfStudents][noOfSubjects];

    for (int i = 0; i < noOfStudents; i++) {
        double total = 0;

        for (int j = 0; j < noOfSubjects; j++) {

           System.out.println("Entering score for student" + (i + 1) + ":");
           System.out.println("Enter score for subjects" + (j + 1) + ":");

           int score;
           do {
              score = scanner.nextInt();
              if (score < 0 || score > 100 ) {
                  System.out.println("Invalid score");
              } 

          } while (score < 0 || score > 100);

       
         scores[i][j] = score;

         total += score;

         System.out.println("Saving >>>>>>>>>>>>>>>>>>>>>>>>>");

         System.out.println("Saved sucessfully");

       }

   }

    System.out.print("=============================================================\nSTUDENT\t"); 

    for(int i = 0; i < noOfSubjects; i++) {
         System.out.printf("\tSUB%-4s", (i+1)  + "");

     }
    
   System.out.print("\tTOT\tAVE\tPOS");

   System.out.print("\n=============================================================\n");

   for(int counter = 0; counter < scores.length; counter++) {

      System.out.print("STUDENT" + (counter+1));

     double total = 0;

     double average = 0;

     for(int k = 0; k < noOfSubjects; k++) {

        System.out.print("\t " + scores[counter][k]);

        total += scores[counter][k];

        average = total / noOfSubjects;
    }

    System.out.printf("    %.2f" , total);

    System.out.printf("    %.2f" , average);

    System.out.println();

    int currentPosition = 1;
    for (int k = 0; k < noOfStudents; k++) {
        if (Arrays.stream(scores[k]).sum() > Arrays.stream(scores[counter]).sum()) {
            currentPosition++;
       }
     }
     System.out.print("\t" + currentPosition + "\n");
   }

   

  System.out.print("================================================================");
int[] subjectsPositions = new int[noOfSubjects];
Arrays.fill(subjectsPositions, 1);
 
for (int j = 0; j < noOfSubjects; j++) {
     int passes = 0;
     int fails = 0;
     for (int i = 0; i < noOfStudents; i++) {
         if (scores[i][j] >= 50) {
             passes++;
           }
         else {
             fails++;
           }
       }
       System.out.println("\nSubject " + (j + 1) + " Summary:");
       System.out.println("Number of Passes: " + passes);
       System.out.println("Number of Fails: " + fails);
       System.out.println("Highest Rank: " + subjectsPositions[j]);

     }
  System.out.print("\n================================================================\n");

}


}